test/lib/ansible_test/_internal/cloud/__init__.py

```diff
@@ -390,14 +407,19 @@ def _generate_resource_prefix():
         :rtype: str
         """
         if is_shippable():
-            return 'shippable-%s-%s' % (
+            prefix = 'shippable-%s-%s' % (
                 os.environ['SHIPPABLE_BUILD_NUMBER'],
                 os.environ['SHIPPABLE_JOB_NUMBER'],
             )
+            return prefix, prefix
 
+        resource_random = random.randint(10000000, 99999999)
         node = re.sub(r'[^a-zA-Z0-9]+', '-', platform.node().split('.')[0]).lower()
 
-        return 'ansible-test-%s-%d' % (node, random.randint(10000000, 99999999))
+        prefix = 'ansible-test-%s-%d' % (node, resource_random)
+        short_prefix = 'ansible-test-%d' % (resource_random)
```

Comment:
It's nice that it's a predictable length, but in tests where I've needed a shorter unique identifier I have just used the unique digits. 64 characters is decent for role names, but some resources have shorter restrictions. I think one of the first places it was added was for load balancer names which can only be 32 characters. And there are other places that use regex_search to find just the digits (such as tags), so I feel like providing the digit string would be more helpful.

Reply:
But I'm not sure. Definitely want @mattclay to weigh in when he's back.

Reply:
The current 'short' string works out as up to 20 characters, which would leave 11 characters for an elb name (assuming you then do -foo ), and has the advantage that it's still clear and easy to spot that this is ansible-test related.  Importantly failures would show up the first time you run the tests rather than working for some people, and not for others.

Given we'll be using this for things like IAM names I think it would still make sense to keep this new short, what would you ( @s-hertel and @mattclay ) think about separately offering that random number?

Reply:
What is the 3rd return value for? I don't see it being used anywhere.

Reply:
@mattclay It was to meet the use cases mentioned above by @s-hertel 

See for example https://github.com/ansible/ansible/blob/devel/test/integration/targets/elb_application_lb/defaults/main.yml where ELB/ALB names need to be kept down to under 32 characters.  I kinda feel the second variable should be sufficient on it's own.

Personally I feel we should encourage folks to consistently use prefix_name / prefix_shortname so that all resources we create are *obviously* related to Ansible testing, and that 10 characters should be enough 'extra' space after the prefix, but I'm no where near as experienced as you guys when it comes to the Ansible test rigs.